### PR TITLE
ci(nightly): add prometheus kind test

### PR DESCRIPTION
**Pull request by adodon2go**
**What type of PR is this?**
test

**Which issue does this PR fix**:
#1930

**What does this PR do / Why do we need it**:

**If an issue # is not available please add repro steps and logs showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades?**

**Does this PR introduce any user-facing change?**:

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/kind.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+name: 'Kind setup'
```
Comment (adodon2go) [inline .github/workflows/kind.yaml:1]: Note: this workflow will be moved to nightly. It is here just during code review to test functionality

**Comment by rchincha** [comment]
Failed ci?

File: examples/kind/kind-ci.sh [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    host: "localhost:${reg_port}"
+    help: "https://kind.sigs.k8s.io/docs/user/local-registry/"
+EOF
+
+## Deploy prometheus operator
+kubectl create -f examples/metrics/kubernetes/prometheus/bundle.yaml
+
+## Deploy the Kubernetes objects for RBAC, prometheus CRD and deploy the service
+kubectl apply -f examples/metrics/kubernetes/prometheus/prom_rbac.yaml
+kubectl apply -f examples/metrics/kubernetes/prometheus/prometheus.yaml
```
Comment (rchincha) [inline examples/kind/kind-ci.sh:61]: also add a metrics user test with authz also

File: examples/kind/kind-ci.sh [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+kubectl apply -f examples/metrics/kubernetes/prometheus/prom_service.yaml
+
+make oci-image
+# copy the image
+skopeo copy --format=oci --dest-tls-verify=false oci:oci docker://localhost:5001/zot-build:latest
```
Comment (rchincha) [inline examples/kind/kind-ci.sh:66]: "zot-build:<commit-id>" instead of zot-build:latest

File: examples/kind/kind-ci.sh [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+  export NO_PROXY=${no_proxy};
+fi
+
+# create registry container unless it already exists
+reg_name='kind-registry'
+reg_port='5001'
+if [ "$(docker inspect -f '{{.State.Running}}' "${reg_name}" 2>/dev/null || true)" != 'true' ]; then
+  docker run \
+    -d --restart=always -p "127.0.0.1:${reg_port}:5000" --name "${reg_name}" \
+    ghcr.io/project-zot/zot-minimal-linux-amd64:latest
```
Comment (rchincha) [inline examples/kind/kind-ci.sh:19]: minimal has no metrics enabled
Just making sure it is **not** this registry metrics that are being uploaded and tested.

Comment (adodon2go) [reply examples/kind/kind-ci.sh:19]: Tested: this is not the case.

**Review by andaaron — APPROVED** [review]
lgtm

**Review by rchincha — APPROVED** [review]
lgtm

**Pull request merged**